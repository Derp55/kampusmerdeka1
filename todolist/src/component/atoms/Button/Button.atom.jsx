import React from "react";

const Button = ({ children, onClick }) => {
  return (
    <button onClick={onClick} style={{ padding: 8, borderRadius: 8 }}>
      {children}
    </button>
  );
};

export default Button;
