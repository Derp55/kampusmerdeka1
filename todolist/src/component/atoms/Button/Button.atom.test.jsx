import { fireEvent, render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import Button from "./Button.atom";

describe("Button", () => {
  it("should render correctly", () => {
    const tree = renderer.create(<Button>tombol</Button>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render children correctly", () => {
    render(<Button>tombol</Button>);
    expect(screen.getByText("tombol")).toBeInTheDocument();
  });

  it("should be able to fire onClick event", () => {
    const onClick = vi.fn();
    render(<Button onClick={onClick}>tombol</Button>);
    fireEvent.click(screen.getByText("tombol"));
    expect(onClick).toHaveBeenCalled();
  });
});
