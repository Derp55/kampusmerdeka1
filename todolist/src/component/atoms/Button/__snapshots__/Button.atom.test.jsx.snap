// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button > should render correctly 1`] = `
<button
  style={
    {
      "borderRadius": 8,
      "padding": 8,
    }
  }
>
  tombol
</button>
`;
